test/config.bats

```diff
@@ -65,3 +65,15 @@ function teardown() {
 	[[ "$RES" == *"crio.runtime.runtimes.runc"* ]]
 	[[ "$RES" == *"crio.runtime.runtimes.crun"* ]]
 }
+
+@test "handle nil workloads" {
+	# when
+	cat << EOF >> "$TESTDIR"/workload.conf
+[crio.runtime.workloads.userns]
+activation_annotation = "io.kubernetes.cri-o.userns-mode"
+allowed_annotations = ["io.kubernetes.cri-o.userns-mode"]
+EOF
+
+	# then
+	"$CRIO_BINARY_PATH" -c "$TESTDIR"/workload.conf -d "" config
```

Comment:
kata-jenkins job fails here, because it cannot find the definition for the default runtime we've set.

`# level=fatal msg="validating runtime config: default_runtime set to \"containerd-shim-kata-v2\", but no runtime entry table [crio.runtime.runtimes.containerd-shim-kata-v2] was found"`

Maybe need to customize the conf file you use with an entry for the default runtime?

Reply:
Another (maybe simpler) suggestion: if the test doesn't imply the runtime, maybe you can unset the CONTAINER_RUNTIMES env variable for this test, like it's done in another test above?

Reply:
good idea! trying that

Reply:
Ok, it didn't help :-(
I just tried locally. The problem is the default runtime, not the list of runtimes.
You need to unset CONTAINER_DEFAULT_RUNTIME for it to work.
I think it's ok to do that in this context, as the runtime is not involved - we're testing the config engine.